src/Pages/Main/Carusel.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+
+import './Carusel.scss';
+
+class Carusel extends React.Component {
+  interval;
+  state = {
+    caruselImgList: [],
+  };
+
+  caruselMovingIndex = (e, index, moveIndex = 0) => {
+    const { caruselImgList } = this.state;
+    const lastIndex = caruselImgList.length;
+
+    index = index
+      ? index
+      : caruselImgList.findIndex(carusel => carusel.isSelected);
+
+    const seletedIndex =
+      index + moveIndex === -1
+        ? lastIndex - 1
+        : index + moveIndex === lastIndex
+        ? 0
+        : index + moveIndex;
+
+    this.setState({
+      caruselImgList: caruselImgList.map((carusel, index) => {
+        index === seletedIndex
+          ? (carusel.isSelected = true)
+          : (carusel.isSelected = false);
+        return carusel;
+      }),
```

Comment:
```suggestion
      carouselImgList: carouselImgList.map((carusel, index) => (
        { ...carousel, isSelected: index === selectedIndex })
      )),
```
이렇게 사용하더라도 같은 결과를 만들 수 있을 것 같네요. 구두로 설명드린 부분은 불변성에 관한 내용이었고, 리액트에서는 가급적 원래 값을 수정하기 보다 새로운 복사본을 만들어 업데이트 한다고 기억하고 계시면 일단 될거 같아요. 조금 더 찾아보시고 이해 안되는 내용 있으시면 또 질문주세요 :)

Reply:
네 기존 배열을 수정하지 않고 복사한 이후 작업하여 재할당 한다.
기억해두겠습니다.